api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -253,7 +299,13 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     if (user == null) {
       workspaces = new ArrayList<>();
     } else {
-      workspaces = workspaceService.dao.findByCreatorOrderByNameAsc(userProvider.get());
+      Set<org.pmiops.workbench.db.model.WorkspaceUserRole> usersWorkspaces;
+      usersWorkspaces = user.getWorkspaceUserRoles();
+      Iterator<org.pmiops.workbench.db.model.WorkspaceUserRole> workspacesIterator = usersWorkspaces.iterator();
+      workspaces = new ArrayList<org.pmiops.workbench.db.model.Workspace>();
+      while(workspacesIterator.hasNext()) {
+        workspaces.add(workspacesIterator.next().getWorkspace());
+      }
```

Comment:
I think this is going to generate a separate SQL `SELECT` for each Workspace a user has access to. We could accept that but maybe leave a TODO on it to optimize if it becomes a bottleneck; or add a new DAO method to do it more directly (select from joining workspaces and permissions, filtering on user ID).

If you want to keep this form, I'd elide some of the intermediate values. Also I think you can skip the fully qualified paths, right -- it's imported above?
```
workspaces = new ArrayList();
if (user != null) {
  for(UserWorkspaceRole workspaceAndRole : user.getWorkspaceUserRoles()) {
    workspaces.add(workspaceAndRole.getWorkspace());
  }
}
```

Reply:
Re: separate SQL statements -- it shouldn't do that. The Workspace will have already been fetched for each UserWorkspaceRole.